appmap/test/web_framework.py

```diff
@@ -363,6 +369,10 @@ def server_url():
 
     @staticmethod
     def record_request_thread():
+        # I've seen occasional test failures, seemingly because the test servers can't handle the
+        # barrage of requests. A tiny bit of delay still causes many, many concurrent requests, but
+        # eliminates the failures.
+        time.sleep(SR.uniform(0, 0.1))
```

Comment:
I ran `poetry run pytest -s -v appmap/test/test_flask.py` while also running saleor tests and got:
```
            # wait for all threads to complete
            for future in concurrent.futures.as_completed(future_to_request_number):
                try:
                    response = future.result()
                except Exception as e:
                    traceback.print_exc()
>               assert response.status_code == 200
E               assert 400 == 200
E                +  where 400 = <Response [400]>.status_code

appmap/test/web_framework.py:410: AssertionError
```

Also ran when not running anything else and got:

```
FAILED appmap/test/test_flask.py::TestRecordRequestsFlask::test_record_request_appmap_enabled_requests_enabled_no_remote - assert 400 == 200
FAILED appmap/test/test_flask.py::TestRecordRequestsFlask::test_record_request_appmap_enabled_requests_enabled_and_remote - assert 400 == 200
FAILED appmap/test/test_flask.py::TestRecordRequestsFlask::test_record_request_appmap_not_enabled_requests_enabled_no_remote - assert 400 == 200
```

Let's keep this change though.  I can imagine it helps. The real problem may be the ability of the tested webserver (Flask/Django) to handle concurrency.

Reply:
The failures I saw turned out to be due to a saleor container using the same port number, 8000.
All tests pass after I stopped this container.  All good.